### DISCUSSION TITLE
collection/PriorityQueue

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#201] [Announcements]
# collection/PriorityQueue

本文详细解析了 Java 优先级队列 PriorityQueue 的实现原理、功能特点以及源码，为您提供了 PriorityQueue 的实际应用示例和性能优化建议。阅读本文，将帮助您更深入地理解 PriorityQueue 在 Java 编程中的应用，从而在实际编程中充分发挥其优势。

https://tobebetterjavaer.com/collection/PriorityQueue.html

**Comment by happier2** [comment]
写的挺好，手工点赞

**Comment by LeeBade** [comment]
原文：“具体来说，remove(Object o)可以分为 2 种情况：

1.删除的是最后一个元素。直接删除即可，不需要调整。
2.删除的不是最后一个元素，从删除点开始以最后一个元素为参照调用一次siftDown()即可。此处不再赘述。”


情况二的完整情况是：删除的不是最后一个元素时，将最后一个元素移动到删除位置，然后先尝试下沉操作。如果下沉后该元素仍在原位置，再尝试上浮操作

另外不传入比较器时总是符合compare语义的最小堆，传入反转顺序的比较器时，是符合compareTo语义的最大堆。代码简化了更好理解，但是好像没有特别指出。希望作者大大百忙之中抽空改改，避免歧义。
